### PR TITLE
CODEOWNERS/MAINTAINERS: Add sysbuild entries

### PR DESCRIPTION
**Pull request by nordicjm**
MAINTAINERS: Add sysbuild entry
    
    Adds a sysbuild maintainers entry.
    
    CODEOWNERS: Add self to sysbuild
    
    Add myself to sysbuild entry.

### CODE REVIEW

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -2362,6 +2362,17 @@ Storage:
...
+Sysbuild:
+  status: maintained
+  maintainers:
+    - tejlmand
+  collaborators:
+    - nordicjm
+  files:
+    - share/sysbuild/
+  labels:
+    - "Sysbuild"
```
Comment (stephanosio) [inline MAINTAINERS.yml:2374]:
```suggestion
    - "area: Sysbuild"
```

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml (hunk 1 above)
Comment (nordicjm) [reply MAINTAINERS.yml]: Fixed

**Review by stephanosio — DISMISSED** [review]

File: MAINTAINERS.yml [hunk 2]
```diff
@@ -2362,6 +2362,17 @@ Storage:
...
     - "area: Storage"
 
+Sysbuild:
+  status: maintained
+  maintainers:
+    - tejlmand
+  collaborators:
+    - nordicjm
+  files:
+    - share/sysbuild/
```
Comment (stephanosio) [inline MAINTAINERS.yml:2372]: would be nice to include relevant sample and test directories as well.

```suggestion
    - share/sysbuild/
    - samples/application_development/sysbuild/
```

Comment (nordicjm) [reply MAINTAINERS.yml:2372]: Added to MAINTAINERS and CODEOWNERS

**Review by stephanosio — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**